test_subscription_from_config is flaky

I.e here https://travis-ci.org/QCoDeS/Qcodes/jobs/450752810 in #1366


```
________________________ test_subscription_from_config _________________________
dataset = test-dataset #1@/tmp/tmplf98kwxq/temp.db
----------------------------------------
x - numeric
y - numeric
basic_subscriber = <function basic_subscriber.<locals>.subscriber at 0x7f74ec0bc400>
    def test_subscription_from_config(dataset, basic_subscriber):
        """
        This test is similar to `test_basic_subscription`, with the only
        difference that another subscriber from a config file is added.
        """
        # This string represents the config file in the home directory:
        config = """
        {
            "subscription":{
                "subscribers":{
                    "test_subscriber":{
                        "factory": "qcodes.tests.dataset.test_subscribing.MockSubscriber"
                        "factory_kwargs":{
                            "lg": false
                        }
                        "subscription_kwargs":{
                            "min_wait": 0
                            "min_count": 1
                            "callback_kwargs": {}
                        }
                    }
                }
            }
        }
        """
        # This little dance around the db_location is due to the fact that the
        # dataset fixture creates a dataset in a db in a temporary directory.
        # Therefore we need to 'backup' the path to the db when using the
        # default configuration.
        db_location = qcodes.config.core.db_location
        with default_config(user_config=config):
            qcodes.config.core.db_location = db_location
    
            assert 'test_subscriber' in qcodes.config.subscription.subscribers
    
            xparam = ParamSpec(name='x', paramtype='numeric', label='x parameter'
                               unit='V')
            yparam = ParamSpec(name='y', paramtype='numeric', label='y parameter'
                               unit='Hz', depends_on=[xparam])
            dataset.add_parameter(xparam)
            dataset.add_parameter(yparam)
    
            sub_id = dataset.subscribe(basic_subscriber, min_wait=0, min_count=1
                                       state={})
            sub_id_c = dataset.subscribe_from_config('test_subscriber')
            assert len(dataset.subscribers) == 2
            assert list(dataset.subscribers.keys()) == [sub_id, sub_id_c]
    
            expected_state = {}
    
            for x in range(10):
                y = -x**2
                dataset.add_result({'x': x, 'y': y})
                expected_state[x+1] = [(x, y)]
>               assert dataset.subscribers[sub_id].state == expected_state
E               assert {1: [(0, 0)]} == {1: [(0, 0)]}
E                 Omitting 1 identical items, use -vv to show
E                 Use -v to get the full diff
tests/dataset/test_subscribing.py:142: AssertionError
```

--- a/qcodes/tests/dataset/test_subscribing.py
+++ b/qcodes/tests/dataset/test_subscribing.py
@@ -14,6 +14,7 @@ from qcodes.tests.dataset.temporary_databases import (
 # pylint: enable=unused-import
 
 from qcodes.tests.test_config import default_config
+from qcodes.tests.common import retry_until_does_not_throw
 
 
 log = logging.getLogger(__name__)
@@ -135,12 +136,20 @@ def test_subscription_from_config(dataset, basic_subscriber):
 
         expected_state = {}
 
-        for x in range(10):
+        # Here we are only testing 2 to reduce the CI time
+        for x in range(2):
             y = -x**2
             dataset.add_result({'x': x, 'y': y})
             expected_state[x+1] = [(x, y)]
-            assert dataset.subscribers[sub_id].state == expected_state
-            assert dataset.subscribers[sub_id_c].state == expected_state
+
+            @retry_until_does_not_throw(
+                exception_class_to_expect=AssertionError, delay=0, tries=10)
+            def assert_expected_state():
+                assert dataset.subscribers[sub_id].state == expected_state
+                assert dataset.subscribers[sub_id_c].state == expected_state
+
+        assert_expected_state()
+
 
 def test_subscription_from_config_wrong_name(dataset):
     """

